src/JointTrajectoryControllerBase.cpp

```diff
@@ -202,10 +211,24 @@ void JointTrajectoryControllerBase::updateStep(const ros::Time& time,
     trajectory->evaluateDerivative(timeFromStart, 2, mDesiredAcceleration);
 
     // TODO: Check path constraints.
-    // TODO: Check goal constraints.
+
+    // Check goal constraints.
+    bool goalConstraintsSatisfied = true;
+    for (const auto& dof : mControlledSkeleton->getDofs()) {
+      for (const auto& pair : mGoalConstraints) {
```

Comment:
I think it'd be more concise to use `find`, rather than this explicit `for` loop through the elements of `mGoalConstraints`.

```cpp
auto it = mGoalConstraints.find(dof->getName());
if (it != mGoalConstraints.end())
{
  auto index = mControlledSkeleton->getIndexOf(dof);
  ...
}
```